### ISSUE TITLE
Confusing javascript root object causes browserify standalone code to malfunction

### ISSUE THREAD

**Issue by ErikLarsson82** [reactjs/React.NET#396]
**TL;DR**: which keyword refers to the global context in the js-engine except for `this`?
`global` exists as a variable on `this` instead of being a proper keyword.

So, we are running this code to render a React component on the server.
`@Html.React("foo", new {})`

The component we are rendering is being bundled by browserify beforehand. Now, browserify has a standalone flag that will expose the component using the following psuedo-code:
```
if window -> window.foo = myComponent; //window doesnt exist, its on the server
if global -> global.foo = myComponent; //global exists, but its a normal variable
else this -> this.foo = myComponent; //so this code doesnt run. if it would, it would work.
```
Since `global` exists as a **variable** on the root object `this` the result is:
`this.global.foo = myComponent; //this is wrong`
what we want:
`this.foo = myComponent; //this is the desired behaviour`

We realise that we can call this code instead
`@Html.React("global.foo", new {})`
... but that causes ReactInitJavascript to attempt to run things on global on the client.

All because of this confusion:
```
global !== this
global === this.global
```

**Comment by dustinsoftware** [comment]
Try bundling with webpack, which has a transform built in already to polyfill `global` in a browser environment.
